File: backend/tests/fixtures/sample_java_repo/src/main/java/Calculator.java
public class Calculator {
    private int value;

    public int add(int a, int b) {
        if (a < 0) {
            return -1;
        }
        return a + b;
    }

    public int divide(int a, int b) {
        if (b == 0) {
            throw new IllegalArgumentException("cannot divide by zero");
        }
        return a / b;
    }
}

File: backend/tests/fixtures/sample_python_repo/calculator.py
def add(a, b):
    """Add two numbers."""
    return a + b


def divide(a, b):
    """Divide a by b, raising ValueError on a zero denominator."""
    if b == 0:
        raise ValueError("cannot divide by zero")
    return a / b


class Calculator:
    def __init__(self, start=0):
        self.value = start

    def add(self, n):
        self.value += n
        return self.value
